src/creatures/combat/combat.cpp

```diff
@@ -427,14 +427,15 @@ ReturnValue Combat::canDoCombat(const std::shared_ptr<Creature> &attacker, const
 					}
 				}
 
-				if (target && target->isSummon() && target->getMaster()->getPlayer()) {
+				if (target && target->isSummon() && target->getMaster() && target->getMaster()->getPlayer()) {
```

Comment:
You can move this for the begin of function:
const auto &targetMaster = target ? target->getMaster() : nullptr;

This will avoid increasing the reference count (as this function is called a lot, it will increase the reference every time it is called).

And you can check this: too: https://github.com/opentibiabr/canary/blob/8b0b0efefa6e61f1b57206f1f9a4a3367421761c/src/creatures/combat/combat.cpp#L406

And remove this: https://github.com/opentibiabr/canary/blob/8b0b0efefa6e61f1b57206f1f9a4a3367421761c/src/creatures/combat/combat.cpp#L410